core/actions/operation.ts

```diff
@@ -156,7 +165,19 @@ export class Operation extends ActionBuilder<dataform.Operation> {
   public dependencies(value: Resolvable | Resolvable[]) {
     const newDependencies = Array.isArray(value) ? value : [value];
     newDependencies.forEach(resolvable => {
-      this.proto.dependencyTargets.push(resolvableAsTarget(resolvable));
+      let dependencyTarget = resolvableAsTarget(resolvable);
+      dependencyTarget.includeDependentAssertions = dependencyTarget.includeDependentAssertions===undefined ? this.dependOnDependencyAssertions : dependencyTarget.includeDependentAssertions;
+      const existingDependencies = this.proto.dependencyTargets.filter(dependency => isSameTarget(dependencyTarget, dependency) && dependency.includeDependentAssertions !== dependencyTarget.includeDependentAssertions)
+      if (existingDependencies.length !== 0){
+          this.session.compileError(
+            `Conflicting "includeDependentAssertions" flag not allowed. Dependency ${dependencyTarget.name} have different value set for this flag.`,
+             this.proto.fileName,
+             this.proto.target
+          )
+          return this;
+      }
+
+      this.proto.dependencyTargets.push(dependencyTarget);
```

Comment:
These blocks are duplicated across action types, we should introduce a shared function for it in `core/actions/index.ts`

Reply:
this function is not used in all sub classes for ActionBuilder. that is why i kept it in operations and tables only.